MQ2GrimGUI.cpp

```diff
@@ -1,505 +1,1510 @@
+			// If they close the window and don't click the button the settings will not be saved and only be temporary.
+			s_WinSettings.showConfigWindow = false;
+			s_SettingModified = false;
+			SaveSettings();
 		}
-		ImGui::SetCursorPos(ImVec2((ImGui::GetCursorPosX() + midX), yPos));
-		ImGui::Text("%d %%", endPercentageInt);
 
-		if (!SplitTargetWindow)
+		if (s_SettingModified)
 		{
-			ImGui::Separator();
-			DrawTargetWindow();
+			ImGui::SameLine();
+			if (ImGui::Button("Revert"))
+			{
+				LoadSettings();
+				s_SettingModified = false;
+			}
 		}
 	}
 	ImGui::End();
 }
 
-static void DrawGroupWindow()
+// Main Window, toggled with /Grimgui command, contains Toggles to show other windows
+static void DrawMainWindow()
 {
+	if (s_WinSettings.showMainWindow)
+	{
+		float displayX = ImGui::GetIO().DisplaySize.x;
+		float displayY = ImGui::GetIO().DisplaySize.y;
+		ImGui::SetNextWindowPos(ImVec2((displayX * 0.5f) - 150, displayY * 0.5f), ImGuiCond_FirstUseEver);
+		ImGui::SetNextWindowSize(ImVec2(300, 200), ImGuiCond_FirstUseEver);
+		if (ImGui::Begin("GrimGUI##MainWindow", &s_WinSettings.showMainWindow))
+		{
+			int sizeX = static_cast<int>(ImGui::GetWindowWidth());
+			int col = 1;
+			col = sizeX / 100;
+			if (col < 1)
+				col = 1;
+
+			if (ImGui::BeginTable("Window List", col))
+			{
+				ImGui::TableNextRow();
+				ImGui::TableNextColumn();
+				for (const auto& option : options)
+				{
+					if (ImGui::Checkbox(option.label, option.setting))
+						SaveSetting(option.setting, &s_SettingsFile[0]);
+					
+					ImGui::TableNextColumn();
+				}
+				ImGui::EndTable();
+			}
+
+			ImGui::Separator();
 
+			if (ImGui::Button("Config"))
+				s_WinSettings.showConfigWindow = true;
 
+		}
+		ImGui::End();
+	}
 }
 
+#pragma endregion
+
+#pragma region Plugin API
 
-// Called periodically by MQ2
 PLUGIN_API void OnPulse()
 {
 	auto now = std::chrono::steady_clock::now();
 	if (GetGameState() == GAMESTATE_INGAME)
 	{
+		s_IsCaster = CheckCaster();
+		s_LockAllWin = s_WinSettings.lockAllWin ? ImGuiWindowFlags_NoMove : ImGuiWindowFlags_None;
+		s_WindowFlags = s_WinSettings.showTitleBars ? ImGuiWindowFlags_None : ImGuiWindowFlags_NoTitleBar;
+
+		if (s_ShowOutOfGame)
+			s_ShowOutOfGame = false; // reset incase we logged back in. =)
+
+		if (!s_CharIniLoaded)
+			UpdateSettingFile();
+
 		if (now - g_LastUpdateTime >= g_UpdateInterval)
 		{
-			g_CharData.Update();  // Refresh character data at the specified interval
-			g_TargetData.Update();
+			if (pAggroInfo)
+			{
+				s_NumSettings.myAggroPct = pAggroInfo->aggroData[AD_Player].AggroPct;
+				s_NumSettings.secondAggroPct = pAggroInfo->aggroData[AD_Secondary].AggroPct;
+				
+				if (pAggroInfo->AggroSecondaryID)
+					s_SecondAggroName = GetSpawnByID(pAggroInfo->AggroSecondaryID)->DisplayedName;
+				else
+					s_SecondAggroName = "Unknown";
+			}
+
+			GetHeading();
+
 			g_LastUpdateTime = now;
+
+			s_DanNetEnabled = mq::IsPluginLoaded("MQ2DanNet");
```

Comment:
use this variable in `DrawGroupWindow`. If you don't want to check _every frame_ that MQ2DanNet is installed, you can listen for it being loaded/unloaded in the OnLoadPlugin/OnUnloadPlugin callbacks.